lib/cdo/code_timer.rb

```diff
@@ -0,0 +1,13 @@
+class CodeTimer
```

Comment:
Also, it would be nice to see a benchmark test verifying the latency hit we accept by instrumenting timing via this class.